### PR TITLE
First draft of Seal HA documentation

### PR DESCRIPTION
**Pull request by sgmiller**
For VAULT-20207

### CODE REVIEW

File: website/content/docs/concepts/seal.mdx [hunk 1]
```diff
@@ -315,3 +315,76 @@ When the quorum of nodes are back up, Raft will elect a leader and the leader
...
+
+1. Edit the Vault configuration, and add the new, target seal configuration.
+1. Restart Vault, allowing re-wrapping to complete.
+1. Edit the Vault configuration, removing the old seal configuration.
+1. Restart Vault, again allowing re-wrapping to complete.
+
+By default, Seal HA is unavailable.  To try the beta, one must set an environment
+variable which enables it:
+
+`SEAL_HA_BETA_ENABLED=true`
```
Comment (divyapola5) [inline website/content/docs/concepts/seal.mdx:351]: I believe the environment variable is VAULT_ENABLE_SEAL_HA_BETA

Comment (sgmiller) [reply website/content/docs/concepts/seal.mdx]: Oops, you're right.  Fixed.

File: website/content/docs/configuration/seal/seal-ha.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+layout: docs
+page_title: Vault Transit - Seals - Configuration
```
Comment (divyapola5) [inline website/content/docs/configuration/seal/seal-ha.mdx:3]: Page title and description needs to be changed

File: website/data/docs-nav-data.json [hunk 3]
```diff
@@ -308,6 +308,9 @@
             "title": "Overview",
             "path": "configuration/seal"
           },
+          { "title": "High Availability (Beta)",
+            "path": "configuration/seal/seal-ha"
```
Comment (stevendpclark) [inline website/data/docs-nav-data.json:312]:
```suggestion
           "badge": {
              "text": "ENT",
              "text": "ENT",
              "type": "outlined",
              "type": "outlined",
              "color": "neutral"
              "color": "neutral"
            },
            "path": "configuration/seal/seal-ha"
```

File: website/content/docs/configuration/seal/seal-ha.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+the order in which to try seals during unseal (least priority first), 
+in the case more than one seal can unwrap a seal wrapped value, the order
+in which to attempt decryption, and which order to attempt to source entropy
+for entropy augmentation.  This can be useful if your seals have different
+performance or cost characteristics.
+
+To add an additional seal, simply add another seal stanza, specifying priority
+and optionally name, and restart Vault.
+
+To remove a seal, remove the corresponding seal stanza and restart.  There must
```
Comment (divyapola5) [inline website/content/docs/configuration/seal/seal-ha.mdx:61]: I think it is not mandatory for having at least one seal remaining. If all are deleted, then it is going to be defaulted to shamir seal right?

Comment (sgmiller) [reply website/content/docs/configuration/seal/seal-ha.mdx:61]: Yes, but at that point it won't be able to unseal or unwrap seal wrapped values, right?

Comment (divyapola5) [reply website/content/docs/configuration/seal/seal-ha.mdx:61]: Yes, you are right. Thanks.

File: website/content/docs/concepts/seal.mdx [hunk 5]
```diff
@@ -315,3 +315,76 @@ When the quorum of nodes are back up, Raft will elect a leader and the leader
...
+
+### Limitations and Known Issues
+
+As Seal HA is in beta, there are certain restrictions that may not be present in 
+the generally available version of the feature:
+
+* Changing seal configurations requires a restart of Vault.
+* Vault must be configured for a single seal at the time of initialization.
+Extra seals can then be added.
+* Upon upgrading to Vault 1.15 or higher, Vault must start with Seal HA disabled
```
Comment (stevendpclark) [inline website/content/docs/concepts/seal.mdx:378]: I don't believe this is quite right, you need to enable the beta flag for a single seal to perform the migration, starting with just the 1.15 binary is not enough.

Comment (sgmiller) [reply website/content/docs/concepts/seal.mdx:378]: Got it.

**Review by divyapola5 — APPROVED** [review]

File: website/content/docs/configuration/seal/seal-ha.mdx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+is the ability to configure more than one seal in order to have resilience against
+outage of a seal service or mechanism.   
+
+Using Seal HA involves configuring extra seals in Vault's server configuration file
+and restarting Vault, after having enabled the Seal HA beta feature by setting
+the environment variable:
+
+`VAULT_ENABLE_SEAL_HA_BETA=true`
+
+Before using Seal HA, one must have upgraded to Vault 1.15 or higher, and started
```
Comment (stevendpclark) [inline website/content/docs/configuration/seal/seal-ha.mdx:29]: Another instance here I believe of talking about the migration needing to occur without the flag set.

File: website/content/docs/configuration/seal/seal-ha.mdx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+<Note title="Seal HA functionality requires Vault Enterprise">
+
+  Seal High Availability requires Vault Enterprise.
+
+</Note>
+
+-> **Warning:** This feature is available as a Beta for evaluation and should not 
+be used in production deployments of Vault.
+
+[Seal High Availability](/vault/docs/content/concepts/seal#Seal-High-Availability)
```
Comment (divyapola5) [inline website/content/docs/configuration/seal/seal-ha.mdx:19]: This link does not work, change it to /vault/docs/concepts/seal#seal-high-availability-enterprise-beta may be?

File: website/content/docs/concepts/seal.mdx [hunk 8]
```diff
@@ -315,3 +315,76 @@ When the quorum of nodes are back up, Raft will elect a leader and the leader
...
+
+Whenever seal configuration changes, Vault must re-wrap all CSPs and seal
+wrapped values, to ensure each value has an entry encrypted by all configured
+seals.  Vault detects these configuration changes automatically, and triggers
+a re-wrap.  Re-wraps can take some time, depending on the number of
+seal wrapped values.  While re-wrapping is in progress, no configuration changes
+to the seals can be made.
+
+Progress of the re-wrap can be monitored using 
+the [`sys/sealwrap/rewrap`](/vault/api-docs/system/seal-rewrap) endpoint.
```
Comment (stevendpclark) [inline website/content/docs/concepts/seal.mdx:368]:
```suggestion
the [`sys/sealwrap/rewrap`](/vault/api-docs/system/sealwrap-rewrap) endpoint.
```

File: website/content/docs/configuration/seal/seal-ha.mdx (hunk 7 above)
Comment (stevendpclark) [inline website/content/docs/configuration/seal/seal-ha.mdx:19]:
```suggestion
[Seal High Availability](/vault/docs/concepts/seal#seal-high-availability-enterprise-betay) 
```

**Review by divyapola5 — APPROVED** [review]
lgtm!

**Review by stevendpclark — APPROVED** [review]
👍

**Pull request merged**